docs/BacwardCompatibility.md

```diff
@@ -0,0 +1,133 @@
+      When this happens, migration instructions are provided. Schema changes may require deleting, editing, and re-creating
+      API objects. The editing process may not be straightforward. The migration may require downtime for applications that
+      rely on the feature.
+- Stable
+    - The version name is vX where X is an integer.
+    - The stable versions of features appear in released software for many subsequent versions.
+### Relations between CRD changes and Fybrik releases
+Based on the above definitions, the relation between CRD versions and Fybrik versions is unidirectional. Which means,
+CRD API changes requires at least minor Fybrik release, on the other hand not each Fybrik release requires a new CRD
+version.
+
+### Upgrade Plan
+The CustomResourceDefinition API provides a workflow for introducing and upgrading to new versions of a 
+CustomResourceDefinition. For more information about 
+[serving multiple versions](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#serving-multiple-versions-of-a-crd) of your CustomResourceDefinition 
+and migrating your objects from one version to another see 
+[Versions in CustomResourceDefinitions](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/)
+
+The first step of adding a new version is to "pick a conversion strategy. Since custom resource objects need to be able 
+to be served at both versions, that means they will sometimes be served at a different version than their storage 
+version. In order for this to be possible, the custom resource objects must sometimes be converted between the version 
+they are stored at and the version they are served at. If the conversion involves schema changes and requires custom 
+logic, a conversion **Webhook** should be used. If there are no schema changes, the default **None** conversion strategy may 
+be used and only the apiVersion field will be modified when serving different versions."
+
+Due to some Fybrik customers' limitations, we cannot use Webhooks in Fybrik deployments. Therefore, below we investigate 
+possible CRD changes without changing its schema.
+
+We were able to change a required field in version v1 to an optional field in v2. In addition, we were able to add a new 
+optional field in v2. This is inline with compatibility examples from [\[5\]](#5)
+
+*Note:* If we go from previous API version to a new one without supporting the previous version, let say `v1alpha1` and 
+a cluster API server has stored objects of the previous version, we will get an error similar to the following:  
+```
+CustomResourceDefinition.apiextensions.k8s.io "plotters.app.fybrik.io" is invalid: status.storedVersions[0]: Invalid value: "v1alpha1": must appear in spec.versions
+```
+
+## Data Catalog and Policy Manager Connector Interfaces
+Fybrik manager communicates with connectors (data catalog and policy manager) over REST API. 
```

Comment:
Aren't we specifically using OpenAPI?  I came across the following relating to versioning with Open API.  https://cloud.google.com/endpoints/docs/openapi/versioning-an-api

Reply:
From OpenAPI [spec](https://swagger.io/specification/)  

> Info Object
The object provides metadata about the API. The metadata MAY be used by the clients if needed, and MAY be presented in editing or documentation generation tools for convenience.

The Info Object contains the API version. 

> REQUIRED. The version of the OpenAPI document 

Our OpenAPI documents [include](https://github.com/fybrik/fybrik/blob/6f78f0d05f40c0767561bac6ccf99411446c51ad/connectors/api/datacatalog.spec.yaml#L3) it as well (`version: 1.0.0`), but we don't use it